src/main/resources/templates/db_scripts/populateEffectiveLocationForExistingItems.sql

```diff
@@ -0,0 +1,15 @@
+UPDATE ${myuniversity}_${mymodule}.item AS it
+SET jsonb = jsonb_set(jsonb, '{effectiveLocationId}',
+	CASE
+	  -- Check item's locations first
+		WHEN jsonb->'temporaryLocationId' IS NOT NULL THEN jsonb->'temporaryLocationId'
+		WHEN jsonb->'permanentLocationId' IS NOT NULL THEN jsonb->'permanentLocationId'
+		-- If no item's locations present - check holding's locations, and return an empty string by default
+		ELSE (
+			SELECT COALESCE(hr.jsonb->'temporaryLocationId', hr.jsonb->'permanentLocationId', to_jsonb(''::text))
```

Comment:
Does this actually set `effectiveLocationId` to an empty string in the jsonb or will COALESCE interpret the empty string as "do not include this field" in the jsonb? If it is an empty string, I don't see how that can be validated with the `uuid` schema by a client. Either it should be a valid UUID or not set at all.

What is actually returned by the API in the default case? Perhaps RMB handles this, which, if it does, seems wrong to me.

Reply:
Hi @mreno-EBSCO, yes, you're completely right. It is fixed now.